### PR TITLE
NO_LOGGING

### PR DESCRIPTION
**Pull request by Srijan1972**
I added a macro NO_LOGGING, which, when defined, will remove all references to the iostream library from the simulator, allowing for a smaller executable size.

### CODE REVIEW

**Comment by romancardenas** [comment]
Hi, Srijan! You are trying to merge this to `Srijan1972-dev-rt2` instead of `dev-rt`. Please, change the destination branch to `dev-rt` so it gets merged into the correct branch.

File: include/cadmium/modeling/devs/component.hpp [hunk 1]
```diff
@@ -109,11 +111,13 @@ namespace cadmium {
 		 * @throws CadmiumModelException if there is no input port with the provided ID.
 		 */
         [[nodiscard]] std::shared_ptr<PortInterface> getInPort(const std::string& id) const {
-            try {
```
Comment (romancardenas) [inline include/cadmium/modeling/devs/component.hpp:112]: Just curious: why did you change the getter functions this way? Is this more performant?

**Review by romancardenas — APPROVED** [review]
LGTM! Before merging it, @Srijan1972 why did you change the getter functions? Is your approach more efficient?

File: include/cadmium/modeling/devs/component.hpp (hunk 1 above)
Comment (Srijan1972) [reply include/cadmium/modeling/devs/component.hpp:112]: This was changed in an attempt to reduce the number of required headers further. The original implementation used the 'out_of_range' exception which required the stdexcept.h header. On making this change only the exception header is needed.

**Comment by romancardenas** [comment]
Merged! Now please open the second PR for RT simulation in your target
